lib/EventListener/UserEditListener.php

```diff
@@ -32,78 +36,72 @@ class UserEditListener implements EventSubscriberInterface
+        Router $router,
+        FormFactory $formFactory
     ) {
         $this->contentTypeService = $contentTypeService;
         $this->contentService = $contentService;
         $this->userController = $userController;
+        $this->router = $router;
+        $this->formFactory = $formFactory;
     }
 
     public static function getSubscribedEvents()
     {
         return [
-            KernelEvents::CONTROLLER => [
-                ['onControllerUserCreate', 20],
-                ['onControllerUserEdit', 30],
+            KernelEvents::REQUEST => [
+                ['beforeUserControllerCreate', -20],
             ],
         ];
     }
 
-    public function onControllerUserCreate(FilterControllerEvent $event)
-    {
-        $request = $event->getRequest();
-
-        if (self::CONTENT_CREATE_ROUTE === $request->attributes->get('_route')) {
-            $routeParams = $request->attributes->get('_route_params', ['contentTypeIdentifier' => null]);
-            $contentTypeIdentifier = $routeParams['contentTypeIdentifier'];
-            if (null !== $contentTypeIdentifier) {
-                try {
-                    $contentType = $this->contentTypeService->loadContentTypeByIdentifier($contentTypeIdentifier);
-                    $isUserBasedContentType = $this->hasUserField($contentType);
-                } catch (NotFoundException $e) {
-                    return;
-                }
-
-                if (!$isUserBasedContentType) {
-                    return;
-                }
-
-                $event->getRequest()->attributes->set('_controller', [$this->userController, 'createAction']);
-                $event->setController([$this->userController, 'createAction']);
-            }
-        }
-    }
-
-    public function onControllerUserEdit(FilterControllerEvent $event)
+    /**
+     * @param \Symfony\Component\HttpKernel\Event\GetResponseEvent $event
+     *
+     * @throws \Exception
+     */
+    public function beforeUserControllerCreate(GetResponseEvent $event)
     {
         $request = $event->getRequest();
+        if (self::CONTENT_BASE_CREATE_ROUTE === $request->attributes->get('_route')) {
+            $form = $this->formFactory->createContent();
+            $form->handleRequest($request);
 
-        if (self::CONTENT_EDIT_ROUTE === $request->attributes->get('_route')) {
-            $routeParams = $request->attributes->get('_route_params', ['contentId' => null]);
-            $contentId = $routeParams['contentId'];
-            if (null !== $contentId) {
-                try {
-                    $content = $this->contentService->loadContent($contentId);
-                    $contentType = $this->contentTypeService->loadContentType($content->contentInfo->contentTypeId);
-                    $isUserBasedContentType = $this->hasUserField($contentType);
-                } catch (NotFoundException $e) {
-                    return;
-                }
+            if ($form->isValid()) {
```

Comment:
`$form->isValid()` is enough, but we mostly using `$form->isSubmitted() && $form->isValid()` for better readability

Reply:
There should be both as afair one just isValid is now deprecated :D